core/src/main/java/org/sonarsource/sonarlint/core/repository/connection/AbstractConnectionConfiguration.java

```diff
@@ -27,26 +27,33 @@ public abstract class AbstractConnectionConfiguration {
    * The id of the connection on the client side
    */
   private final String connectionId;
+  private final boolean disableNotifications;
 
-  protected AbstractConnectionConfiguration(String connectionId) {
+  protected AbstractConnectionConfiguration(String connectionId, boolean disableNotifications) {
     Objects.requireNonNull(connectionId, "Connection id is mandatory");
     this.connectionId = connectionId;
+    this.disableNotifications = disableNotifications;
   }
 
   public String getConnectionId() {
     return connectionId;
   }
 
+  public boolean isDisableNotifications() {
+    return disableNotifications;
+  }
+
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     var that = (AbstractConnectionConfiguration) o;
-    return Objects.equals(connectionId, that.connectionId);
+    return Objects.equals(connectionId, that.connectionId)
+      && Objects.equals(disableNotifications, that.disableNotifications);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(connectionId);
+    return Objects.hash(connectionId, disableNotifications);
```

Comment:
Doesn't look very useful to have the disableNotifications part of the hash, since connectionId should be unique anyway.